py-polars/polars/dataframe/frame.py

```diff
@@ -7249,131 +7249,174 @@ def pivot(
         separator: str = "_",
     ) -> Self:
         """
-        Create a spreadsheet-style pivot table as a DataFrame.
+        Pivot this `DataFrame` from long to wide format. The inverse of :func:`melt`.
 
-        Only available in eager mode. See "Examples" section below for how to do a
-        "lazy pivot" if you know the unique column values in advance.
+        `pivot` transforms a "long-format" `DataFrame`, where each row represents an
+        observation, into a "wide-format" one, where each element represents an
+        observation.
+
+        `pivot` is only available in eager mode, since the schema of the output
+        `DataFrame` depends on the number of unique values in `columns`. However, if you
+        know these unique values in advance, you can perform a "lazy pivot", as shown in
+        the "Examples" section below.
+
+        Warnings
+        --------
+        `pivot` will introduce `null` values in the output `DataFrame` if not every
+        combination of the values in `index` and `columns` appears in the input
+        `DataFrame`.
 
         Parameters
         ----------
         values
-            Column values to aggregate. Can be multiple columns if the *columns*
-            arguments contains multiple columns as well. If None, all remaining columns
-            will be used.
+            The column(s) whose values will become the values of the output `DataFrame`.
+            If None, all columns not specified in the index and columns arguments will
+            be used.
         index
-            One or multiple keys to group by.
+            The column(s) whose values will become the index of the output `DataFrame`.
         columns
-            Name of the column(s) whose values will be used as the header of the output
-            DataFrame.
+            The column(s) whose values will become the columns of the output
+            `DataFrame`. If these columns are not :class:`String` columns, their values
+            will be coerced to strings, since polars column names must be strings.
         aggregate_function
-            Choose from:
-
-            - None: no aggregation takes place, will raise error if multiple values are in group.
-            - A predefined aggregate function string, one of
-              {'min', 'max', 'first', 'last', 'sum', 'mean', 'median', 'len'}
-            - An expression to do the aggregation.
+            A function to aggregate multiple `values` with the same `index` and
+            `columns` prior to pivoting, equivalent to using :func:`group_by` as a
```

Comment:
does `group_by` render correctly here? you may need to use 
```
:meth:`DataFrame.group_by`
```
?

Reply:
Yes,
```python
:func:`group_by`
```
works - see how https://github.com/pola-rs/polars/blob/py-0.20.13/py-polars/polars/dataframe/frame.py#L7164 renders correctly at https://docs.pola.rs/py-polars/html/reference/dataframe/api/polars.DataFrame.fill_nan.html.